api4/channel.go

```diff
@@ -282,7 +282,7 @@ func updateChannelPrivacy(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	props := model.StringInterfaceFromJson(r.Body)
-	privacy, ok := props["privacy"].(string)
+	privacy, ok := props["privacy"].(model.ChannelType)
```

Comment:
Here we probably need to first assert the value in the map is a `string` and then do a conversion to `model.ChannelType`. Unless of course we start sending a proper structured type instead of a generic map.